lib/SPIRV/libSPIRV/SPIRVType.h

```diff
@@ -249,7 +249,8 @@ class SPIRVTypePointer : public SPIRVType {
     auto Cap = getVec(CapabilityAddresses);
     if (getElementType()->isTypeFloat(16))
       Cap.push_back(CapabilityFloat16Buffer);
-    auto C = getCapability(ElemStorageClass);
+    auto C = getAllEnablingCapabilities(ElemStorageClass);
+    Module->chooseBestCapability(C);
```

Comment:
```suggestion
    Module->chooseBestCapability(getAllEnablingCapabilities(ElemStorageClass));
```

Reply:
There is "Cap.insert(Cap.end(), C.begin(), C.end());" one line below, so we cannot optimize it this way